### DISCUSSION TITLE
Add phone number to MultiFactorInfo hint for multi-factor

### DISCUSSION THREAD

**Discussion by huwmartin** [invertase/react-native-firebase#7190] [Ideas]
Firebase iOS and Android SDKs support a `phoneNumber` property on a `MultiFactorInfo` `hint` (see [here](https://firebase.google.com/docs/reference/js/v8/firebase.auth.phonemultifactorinfo#phonenumber) and [here](https://firebase.google.com/docs/auth/ios/multi-factor)), this shows the phone number used for MFA when authenticated and is partially masked when the user is signing in with MFA, this would be quite nice to be able to show "We've sent an SMS to +******461". We've patched this into @react-native-firebase/auth in our own project but I wanted to see whether there was an appetite for this.

```patch
diff --git a/node_modules/@react-native-firebase/auth/android/src/main/java/io/invertase/firebase/auth/ReactNativeFirebaseAuthModule.java b/node_modules/@react-native-firebase/auth/android/src/main/java/io/invertase/firebase/auth/ReactNativeFirebaseAuthModule.java
index 62f928b..4858b79 100644
--- a/node_modules/@react-native-firebase/auth/android/src/main/java/io/invertase/firebase/auth/ReactNativeFirebaseAuthModule.java
+++ b/node_modules/@react-native-firebase/auth/android/src/main/java/io/invertase/firebase/auth/ReactNativeFirebaseAuthModule.java
@@ -2206,6 +2206,10 @@ class ReactNativeFirebaseAuthModule extends ReactNativeFirebaseModule {
     hintMap.putString("factorId", hint.getFactorId());
     hintMap.putString("uid", hint.getUid());
 
+    if (PhoneMultiFactorGenerator.FACTOR_ID.equals(hint.getFactorId())) {
+      hintMap.putString("phoneNumber", ((PhoneMultiFactorInfo) hint).getPhoneNumber());
+    }
+
     return hintMap;
   }
 
diff --git a/node_modules/@react-native-firebase/auth/ios/RNFBAuth/RNFBAuthModule.m b/node_modules/@react-native-firebase/auth/ios/RNFBAuth/RNFBAuthModule.m
index 8f410ec..908ba74 100644
--- a/node_modules/@react-native-firebase/auth/ios/RNFBAuth/RNFBAuthModule.m
+++ b/node_modules/@react-native-firebase/auth/ios/RNFBAuth/RNFBAuthModule.m
@@ -1453,6 +1453,7 @@ - (NSDictionary *)firebaseUserToDict:(FIRUser *)user {
       @"uid" : hint.UID,
       @"factorId" : [self getJSFactorId:(hint.factorID)],
       @"displayName" : hint.displayName == nil ? [NSNull null] : hint.displayName,
+      @"phoneNumber" : hint.phoneNumber == nil ? [NSNull null] : hint.phoneNumber,
       @"enrollmentDate" : enrollmentDate,
     }];
   }
```

Happy to open a PR for this, open to discussion on any amendments on how you would like to see this implemented. Main thing I could use some steer on is how this should be expressed for the TypeScript types - for the patch within our project we simply added `phoneNumber` as an optional property on `MultiFactorInfo`, but this could be split into some union type like `PhoneMultiFactorInfo | MultiFactorInfo` to more closely mirror the native SDKs.

```patch
diff --git a/node_modules/@react-native-firebase/auth/lib/index.d.ts b/node_modules/@react-native-firebase/auth/lib/index.d.ts
index 30d2beb..6f57ad1 100644
--- a/node_modules/@react-native-firebase/auth/lib/index.d.ts
+++ b/node_modules/@react-native-firebase/auth/lib/index.d.ts
@@ -431,6 +431,10 @@ export namespace FirebaseAuthTypes {
      * User friendly name for this factor.
      */
     displayName?: string;
+    /**
+     * The phone number associated with the current second factor.
+     */
+    phoneNumber?: string;
     /**
      * Time the second factor was enrolled, in UTC.
      */
```

**Comment by BlindDev** [comment]
We have the same requirements for our app. Would be really nice to have It. Maybe it worth opening PR. Thanks for the patch BTW.

**Comment by yuvalhermelin-fijoya** [comment]
@paulb777 @mikehardy What do you think? Can I or @huwmartin (if he wants/is available) open a PR for this? It would be nice to have access to the last four digits of the phone number for UI purposes before the user completes MFA verification. Thanks!

**Reply by mikehardy** [reply]
Hey @yuvalhermelin-fijoya I'm sorry I did not see this! I groom the issues list pretty thoroughly but missed this discussion when it came in. In general we aim to 1) support everything the JS APIs have, in a JS API compatible format and 2) even beyond that if the native iOS/Android SDKs support something the JS API doesn't, expose those things in a way that is as JS-API-compatible as possible

So yes! There is definitely appetite for a PR here and I'd be happy to help get it merged+released

**Reply by mikehardy** [reply]
Also a quick 👋 and thanks to @huwmartin for noticing and posting up what looks like a functional patch although the line numbers appear to have moved around since June, the patch looks pretty good

**Reply by yuvalhermelin-fijoya** [reply]
Thanks, I'll wait for #7565 as per your suggestion in #7189  and hope it fixes it :) If not I'll open a PR
